Sources/PrimerSDK/Classes/User Interface/Apple Pay/ApplePayViewModel.swift

```diff
@@ -1,69 +1,317 @@
+                            let state: AppStateProtocol = DependencyContainer.resolve()
+
+                            guard let applePayConfigId = self?.applePayConfigId else {
+                                return completion(PaymentException.missingConfigurationId)
+                            }
+
+                            let instrument = PaymentInstrument(
+                                paymentMethodConfigId: applePayConfigId,
+                                token: applePayPaymentResponse.token,
+                                sourceConfig: ApplePaySourceConfig(source: "IN_APP", merchantId: merchantIdentifier)
+                            )
+                            
+                            applePayService.tokenize(instrument: instrument) { [weak self] (result) in
+                                switch result {
+                                case .failure(let err):
+                                    // Dismiss and show error screen
+                                    ErrorHandler.shared.handle(error: err)
+                                    DispatchQueue.main.async {
+                                        Primer.shared.presentingViewController?.dismiss(animated: true, completion: {
+                                            if !settings.hasDisabledSuccessScreen {
+                                                Primer.shared.root = RootViewController()
+                                                let router: RouterDelegate = DependencyContainer.resolve()
+                                                router.setRoot(Primer.shared.root!)
+                                                router.show(.error(error: err))
+                                                Primer.shared.presentingViewController?.present(Primer.shared.root!, animated: true)
+                                            }
+                                        })
+                                        
+                                        Primer.shared.delegate?.checkoutFailed(with: err)
+                                    }
+                                    completion(err)
+                                    
+                                case .success(let token):
+                                    DispatchQueue.main.async {
+                                        if Primer.shared.flow.internalSessionFlow.vaulted {
+                                            Primer.shared.delegate?.tokenAddedToVault(token)
+                                            completion(nil)
+                                        } else {
+                                            //settings.onTokenizeSuccess(token, completion)
+                                            Primer.shared.delegate?.authorizePayment(token, { (err) in
+                                                DispatchQueue.main.async {
+                                                    Primer.shared.presentingViewController?.dismiss(animated: true, completion: {
```

Comment:
Callback heeeeeell! :D

Reply:
Yep :( 

We do need to assign some time to embed PromiseKit